### PR TITLE
Plans 2023: Split and export the individual grids (FeaturesGrid, ComparisonGrid) - take 2

### PR DESCRIPTION
**Pull request by chriskmnds**
Related to https://github.com/Automattic/wp-calypso/issues/78266
Takes over https://github.com/Automattic/wp-calypso/pull/80915

## Proposed Changes

Converts the main package component `plans-grid` into an index of individual parts, exported and assembled at the consuming end. We effectively now export `FeaturesGrid` and `ComparisonGrid` separately, and a single `gridPlans` prop in place of `gridPlansForFeauresGrid`, etc.

### Additional Notes

- There is some repetition in the "wrappers" exported from the main index file, and that's intentional. All of that will gradually shrink to simply reflected imports as we clean up more dependencies i.e. there will be no need for these wrappers altogether.
- The `ComparisonGrid` carries its title along, which is something incredibly small that we can handle in a follow-up. 
- A props refactor should also happen on top of this PR. A separate issue exists: https://github.com/Automattic/wp-calypso/issues/82335
- Some performance concerns are being handled in a separate PR. For the most part, things should be functioning reasonably.

## Testing Instructions

- Confirm `/start` and `/plans/[ paid | free ]` render as before
- Confirm tailored flows (newsletter, wooexpress) work as before
- Confirm plan-term toggle works correctly

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

File: client/my-sites/plans-grid/index.tsx [hunk 1]
```diff
@@ -24,9 +26,19 @@ import type { MinimalRequestCartProduct } from '@automattic/shopping-cart';
...
+ *	TODO clk: On PlansGridProps:
+ *	1.	These props need to be split into what's needed separately for the internal grids and what's needed for the exported wrappers.
+ *		This is currently also used as the internal type for the FeaturesGrid (wrongly).
+ *			- onUpgradeClick is only needed for the wrappers exported from here. Internally the grids take a transformed
+ *			handleUpgradeClick/onUpgradeClick function (force renamed to handleUpgradeClick in FeaturesGrid)
+ *			- allFeaturesList is only relevant for the ComparisonGrid
+ *			- gridPlanForSpotlight is only relevant for the FeaturesGrid
+ *			- etc.
+ *	2.	Explicitly set optional props and default values. Only absolutely vital props should be required.
+ */
```
Comment (chriskmnds) [inline client/my-sites/plans-grid/index.tsx:39]: This we ought to address in a bit of immediate follow-up. We may consider splitting these props across several pieces, and updating the respective READMEs while at it, possibly:

1. `ContextProps` (not exported) all the props that pertain to things added to the context
2. `SharedPlansGridProps` (not exported) all the props that are shared between the FeaturesGrid and ComparisonGrid (et al)
3. `ComparisonGridProps`
4. `FeaturesGridProps`

Comment (chriskmnds) [reply client/my-sites/plans-grid/index.tsx:39]: Created a separate issue for this https://github.com/Automattic/wp-calypso/issues/82335

**Comment by chriskmnds** [comment]
> Some performance concerns are being handled in a separate PR. For the most part, things should be functioning reasonably.

Performance optimisations addressed in https://github.com/Automattic/wp-calypso/pull/82403

**Review by southp — APPROVED** [review]
Following [the test plan I previously wrote](https://github.com/Automattic/wp-calypso/pull/79733#issuecomment-1651192466) it all works as expected :) On top of that test plan, I've also tried out the storage add-on dropdown, term toggle, plan picker of the comparison grid, and the mobile resolutions. Oh my, we really should compile a QA guide or something 🙃 

I do notice there are some things broken in the comparison grid in `/plans`, e.g. the styling of the Free plan is missing, the tooltips are flashing, "show features" requires 3 clicks to be expanded. However, those also happen in production so should be tracked and resolved separately from this 🚢

**Comment by chriskmnds** [comment]
Thanks @southp .

> I do notice there are some things broken in the comparison grid in /plans, e.g. the styling of the Free plan is missing, the tooltips are flashing, "show features" requires 3 clicks to be expanded. However, those also happen in production so should be tracked and resolved separately from this 🚢

I don't notice these in the live image 🤔 

Do you mind confirming and we can create a few issues for them?

> "show features" requires 3 clicks to be expanded.

This sounds pretty severe. Do you mean the "show all features" toggle in the mobile cards?

**Pull request merged**